runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -445,6 +445,149 @@ TR::Register * caseConversionHelper(TR::Node* node, TR::CodeGenerator* cg, bool
+   TR::Register* indexRegister = cg->allocateRegister();
+   TR::Register* offsetAddress = cg->allocateRegister();
+   TR::Register* scratch = offsetAddress;
+
+   TR::Register* charBufferVector = cg->allocateRegister(TR_VRF);
+   TR::Register* resultVector = cg->allocateRegister(TR_VRF);
+   TR::Register* valueVector = cg->allocateRegister(TR_VRF);
+
+   TR::LabelSymbol* loopLabel = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* fullVectorLabel = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* notFoundInResidue = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* foundLabel = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* foundLabelExtractedScratch = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* failureLabel = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* doneLabel = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+
+   bool isCompressed = node->getSymbol()->castToMethodSymbol()->getMethod()->parmType(1) == TR::Int8;
+   const int elementSizeMask = isCompressed ? 0x0 : 0x1;   // byte or halfword mask
+   uintptrj_t headerSize = TR::Compiler->om.contiguousArrayHeaderSizeInBytes();
+   const int8_t sizeOfVector = cg->machine()->getVRFSize();
+   const bool is64 = TR::Compiler->target.is64Bit();
+
+   TR::RegisterDependencyConditions * regDeps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 8, cg);
+   regDeps->addPostCondition(address, TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(loopCounter, TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(indexRegister, TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(loadLength, TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(offsetAddress, TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(charBufferVector, TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(resultVector, TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(valueVector, TR::RealRegister::AssignAny);
+
+   generateVRRfInstruction(cg, TR::InstOpCode::VLVGP, node, valueVector, index, value);
+   generateVRIcInstruction(cg, TR::InstOpCode::VREP, node, valueVector, valueVector, (sizeOfVector / (1 << elementSizeMask)) - 1, elementSizeMask);
+   generateVRIaInstruction(cg, TR::InstOpCode::VGBM, node, resultVector, 0, 0);
+   generateVRIaInstruction(cg, TR::InstOpCode::VGBM, node, charBufferVector, 0, 0);
+
+   if (is64)
+      {
+      generateRREInstruction(cg, TR::InstOpCode::LLGFR, node, indexRegister, index);
+      } 
+   else 
+      {
+      generateRRInstruction(cg, TR::InstOpCode::LR, node, indexRegister, index);
+      }
+   generateRRInstruction(cg, TR::InstOpCode::SR, node, size, index);
+
+   if (!isCompressed)
+      {
+      generateRSInstruction(cg, TR::InstOpCode::SLLK, node, size, size, 1);
+      generateRSInstruction(cg, TR::InstOpCode::SLLK, node, indexRegister, indexRegister, 1);
```

Comment:
Can we use `SLL` here instead since the source and target registers are the same? `SLL` is a 4 byte instruction while `SLLK` is a 6-byte.

Reply:
1c7e1cd